app/views/editions/_actions.html.erb

```diff
@@ -4,10 +4,14 @@
 
       <% edition_policy = EditionPolicy.new(current_user, edition) %>
 
-      <% if edition_policy.can_preview? %>
+      <% if edition_policy.can_preview? && edition_policy.can_discard_draft? %>
```

Comment:
It would make sense to me if the logic went into the `#can_preview?` method. Keeping the confusion of conditions out of the view is a good thing.

I get where Ollie is going though. Whether a draft can be discarded or not doesn't seem directly related to whether it can be previewed. Perhaps a private method would make it clearer? Eg.

```
def can_discard_draft?
    is_being_edited?
  end

  def can_preview?
    edition.persisted? && is_being_edited?
  end

private

  def is_being_edited?
    !Edition::STATES_THAT_UPDATE_THE_FRONTEND.include?(edition.state)
  end
```